gax/retry_policy.h

```diff
@@ -48,62 +51,112 @@ class RetryPolicy {
    * @return true if the RPC operation should be retried.
    */
   virtual bool OnFailure(Status const& status) = 0;
+
+  /**
+   * Calculate the deadline for the next RPC operation.
+   *
+   * Any internal state modification, if neccessary, should occur in OnFailure.
+   *
+   * Note: this is different from the deadline in LimitedDurationRetryPolicy,
+   *       which is the deadline after which retry attempts should be abandoned.
+   *
+   * @return the _deadline_ for the next RPC, NOT its maximum _duration_.
+   */
+  virtual std::chrono::system_clock::time_point OperationDeadline() const = 0;
+};
+
+class DefaultClock {
```

Comment:
Shouldn't this class get a Doxygen comment?

Reply:
Aaah, too quick off the starting line :P

I'll add it in the next PR.